torch/onnx/symbolic_opset12.py

```diff
@@ -209,3 +209,48 @@ def unfold(g, input, dimension, size, step):
         return squeeze
     else:
         return _unimplemented("Unfold", "input size not accessible")
+
+@parse_args('v', 'v', 'is', 'is', 'v')
+def tensordot(g, input_a, input_b, dims_a, dims_b, out=None):
+    if out is not None:
+        _unimplemented("Tensordot", "Out parameter is not supported for tensordot.")
+
+    from torch.onnx.symbolic_opset9 import view, permute
+    for i in range(len(dims_a)):
+        if sym_help._get_tensor_dim_size(input_a, dims_a[i]) != sym_help._get_tensor_dim_size(input_b, dims_b[i]):
+            raise RuntimeError("Dimensions are not matched for operator tensordot: a - {}, b - {}.".format(dims_a, dims_b))
+
+    dim_count_a = input_a.type().dim()
```

Comment:
I think there is helper function for getting rank in `sym_help` as well.

Also for this implementation, it would seem the tensor rank is required to be known as static number. This comes from limitation of potential `transpose/permute` later, which requires static rank, as well as from the code below for constructing dimensions. In that case, it would be better to throw not supported error when rank is unknown. (However, if `transpose` is not used, then we can lift the requirement on rank, by doing the below construction on dynamic shape with nodes of `ONNX::Shape`, and `ONNX::Slice`)

Reply:
Update code to infer the size.